Unit Tests/Helpers/String+Extensions.swift

```diff
@@ -1,6 +1,6 @@
 import XCTest
 
-extension String {
+extension String: Error {
```

Comment:
I have an issue with this change as well. It's part of the PR only because the old draft of token providers defined `extension String: Error`. I deleted that and our unit tests stopped to compile due to the fact that `toData()` method throws a string. I think this should be improved, but I wasn't sure if that should be part of this PR.

Reply:
I think we possibily need to have a chat about what types of error we might expose in the public interface (and internally as well TBH).  This may change as a result, so leave it for now and it might change in the future 👍